chia/util/network.py

```diff
@@ -145,30 +147,40 @@ def class_for_type(type: NodeType) -> Any:
     raise ValueError("No class for type")
 
 
-def get_host_addr(host: str, *, prefer_ipv6: bool = False) -> IPAddress:
+async def _resolve_inner(host: str, resolver: Resolver, rd_type: RdataType) -> IPAddress:
+    try:
+        answer = await resolver.resolve(qname=host, rdtype=rd_type)
+    except Exception as e:
+        raise NetworkError(f"Resolve {rd_type.name} failed for {host}: {e}") from e
+    if answer.rrset is None or len(answer.rrset) == 0:
+        raise NetworkError(f"Empty {rd_type.name} answer for {host}")
+    try:
+        return IPAddress.create(str(answer.rrset[0]))
+    except Exception as e:
+        raise NetworkError(f"Invalid {rd_type.name} response for {host}: {answer.rrset[0]}") from e
+
+
+async def resolve(host: str, *, prefer_ipv6: bool = False) -> IPAddress:
+    if host == "localhost":
```

Comment:
might be useful to add in `ip6-localhost` which is linux (possibly Ubuntu) specific, but is common on those systems. Also on Ubuntu, `localhost` will always resolve to the IPv4 loopback afaict, so this may be a not so subtle change in behaviour

Reply:
Worth testing if this limitation of asyncresolver is specific to localhost , but doesn't impact any other items in the `/etc/hosts` file?